### PR TITLE
feat:reset password

### PR DESCRIPTION
**Pull request by yang1666204**
## Summary

1. Support (new) users to reset their passwords
3. Fixed some interaction and UI issues with permission control.
4. Update API.
image

### CODE REVIEW

File: ui/src/api/index.ts [hunk 1]
```diff
@@ -28,7 +28,16 @@ globalAxios.interceptors.response.use(
...
       location.href = '/#/login';
     } else {
-      message.error(error?.response?.data?.message || error.message);
+      const { response } = error;
+      if (
+        response?.status === 400 &&
+        response?.config?.url?.split('/')?.pop() === 'password' &&
+        response?.data?.message === 'Error BadRequest: password is incorrect'
+      ) {
+        message.error('原密码输入不正确')
```
Comment (powerfooI) [inline ui/src/api/index.ts:37]: The specific judgement for incorrect password here is not so proper. Maybe putting them in another file named like `errorHandling.ts` is better.

File: ui/src/components/customModal/HandleAccountModal.tsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { useRequest } from 'ahooks';
 import { Form, Input, Select, message } from 'antd';
 import { omit } from 'lodash';
 import { useEffect, useMemo } from 'react';
+import { usePublicKey,encryptText } from '@/hook/usePublicKey';
```
Comment (powerfooI) [inline ui/src/components/customModal/HandleAccountModal.tsx:8]: There should be a space here.

File: ui/src/components/customModal/HandleAccountModal.tsx [hunk 3]
```diff
@@ -41,6 +43,7 @@ export default function HandleAccountModal({
   const onFinish = async (
     formData: AcCreateAccountParam & { confirmPassword: string },
   ) => {
+    formData.password = encryptText(formData.password,publicKey) as string;
```
Comment (powerfooI) [inline ui/src/components/customModal/HandleAccountModal.tsx:46]: Format this file.

File: ui/src/components/customModal/ResetPwdModal.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                }
+                return Promise.resolve();
+              },
+            }),
+          ]}
+          label="新密码"
+          name={'password'}
+        >
+          <Input type="password" placeholder="请输入" />
+        </Form.Item>
```
Comment (powerfooI) [inline ui/src/components/customModal/ResetPwdModal.tsx:80]: There should be an another input named like `PasswordConfirm` for confirming password that the user inputs.

File: ui/src/pages/Layouts/index.tsx [hunk 5]
```diff
@@ -2,21 +2,25 @@ import { ConfigProvider } from '@oceanbase/design';
 import { getAppInfoFromStorage } from '@/utils/helper';
 import enUS from '@oceanbase/ui/es/locale/en-US';
 import zhCN from '@oceanbase/ui/es/locale/zh-CN';
-import { Outlet, getLocale, useNavigate } from '@umijs/max';
+import { Outlet, getLocale, useNavigate, useModel,history } from '@umijs/max';
```
Comment (powerfooI) [inline ui/src/pages/Layouts/index.tsx:5]: Format this file

**Review by powerfooI — CHANGES_REQUESTED** [review]

File: ui/src/components/customModal/ResetPwdModal.tsx (hunk 4 above)
Comment (yang1666204) [reply ui/src/components/customModal/ResetPwdModal.tsx:80]: Here is to change the password, `PasswordConfirm` is in the `./pages/ResetPwd` directory

Comment (powerfooI) [reply ui/src/components/customModal/ResetPwdModal.tsx:80]: Yes I know. User needs to confirm the password he input just now.

Comment (yang1666204) [reply ui/src/components/customModal/ResetPwdModal.tsx:80]: okey.

**Review by powerfooI — APPROVED** [review]
LGTM

**Pull request merged**